coffea/jetmet_tools/CorrectedMETFactory.py

```diff
@@ -0,0 +1,161 @@
+from coffea.jetmet_tools.JECStack import JECStack
+import awkward
+import awkward1
+import numpy as np
+import warnings
+from copy import copy
+
+
+class CorrectedMETFactory(object):
+
+    def __init__(self, name_map):
+        if 'xMETRaw' not in name_map or name_map['xMETRaw'] is None:
+            warnings.warn('There is no name mapping for ptRaw,'
+                          ' CorrectedJets will assume that <object>.x is raw pt!')
+            name_map['xMETRaw'] = name_map['METx'] + '_raw'
+        self.treat_pt_as_raw = 'ptRaw' not in name_map
+
+        if 'yMETRaw' not in name_map or name_map['yMETRaw'] is None:
+            warnings.warn('There is no name mapping for massRaw,'
+                          ' CorrectedJets will assume that <object>.x is raw pt!')
+            name_map['yMETRaw'] = name_map['METy'] + '_raw'
+
+        self.name_map = name_map
+
+    def build(self, MET, corrected_jets, lazy_cache):
+        if lazy_cache is None:
+            raise Exception('CorrectedMETFactory requires a awkward-array cache to function correctly.')
```

Comment:
Same option here to use `jets.behavior["__events_factory__"]._cache` if the key exists.